### PR TITLE
Fix test

### PR DESCRIPTION
**Pull request by octaviospain**
I moved the `PersonImpl` to a separate class as it was in the beginning, to keep this simple.

### CODE REVIEW

File: src/test/java/bb/example/PersonDatabaseTest.java [hunk 1]
```diff
@@ -1,42 +1,52 @@
...
-import static org.mockito.Mockito.verify;
-import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.*;
 
+@ExtendWith(MockitoExtension.class)
 class PersonDatabaseTest {
+
     @Mock
-    PersonI mockedPerson;
-    @InjectMocks
```
Comment (octaviospain) [inline src/test/java/bb/example/PersonDatabaseTest.java]: `@InjectMocks` is for something different. Here you are testing behavior on PersonDatabase class, so you don't mock it, you create an instance of it on each test case

File: src/test/java/bb/example/PersonDatabaseTest.java [hunk 2]
```diff
@@ -1,42 +1,52 @@
...
-        PersonI addedPerson = database.findById(1);
-
-        assertThat(addedPerson).isNotNull();
-        assertThat(addedPerson.getName()).isEqualTo("John");
-        assertThat(addedPerson.getSurname()).isEqualTo("Doe");
-        assertThat(addedPerson.getPhone()).isEqualTo("123456789");
-        assertThat(addedPerson.getAge()).isEqualTo(25);
+    @DisplayName("Should add person to database")
+    void shouldAddPersonTest() {
+        when(person.getId()).thenReturn(1);
```
Comment (octaviospain) [inline src/test/java/bb/example/PersonDatabaseTest.java:29]: You specify the expected state of the mock first

File: src/test/java/bb/example/PersonDatabaseTest.java [hunk 3]
```diff
@@ -1,42 +1,52 @@
...
+        when(person.getName()).thenReturn("John");
+        when(person.getSurname()).thenReturn("Doe");
+        when(person.getPhone()).thenReturn("123456789");
+        when(person.getAge()).thenReturn(25);
+
+        PersonI foundPerson = database.findById(1);
+        assertThat(foundPerson).isNull();
```
Comment (octaviospain) [inline src/test/java/bb/example/PersonDatabaseTest.java:36]: For better understanding of a test, you can show the behavior before what you attempt to do, in this case is too check that a person with id `1` exists.

File: src/test/java/bb/example/PersonDatabaseTest.java [hunk 4]
```diff
@@ -1,42 +1,52 @@
...
+        foundPerson = database.findById(1);
+
+        assertThat(foundPerson).isNotNull();
+        assertThat(foundPerson.getId()).isEqualTo(1);
+        assertThat(foundPerson.getName()).isEqualTo("John");
+        assertThat(foundPerson.getSurname()).isEqualTo("Doe");
+        assertThat(foundPerson.getPhone()).isEqualTo("123456789");
+        assertThat(foundPerson.getAge()).isEqualTo(25);
+
+        verify(person).setId(1);
```
Comment (octaviospain) [inline src/test/java/bb/example/PersonDatabaseTest.java:49]: Here you verify that the method `setId` was called on the mock`

File: src/test/java/bb/example/PersonDatabaseTest.java [hunk 5]
```diff
@@ -1,42 +1,52 @@
...
+        verifyNoMoreInteractions(person);
```
Comment (octaviospain) [inline src/test/java/bb/example/PersonDatabaseTest.java:50]: If there were more operations called on the mock, the test will fail in this line

**Review by poLLib — APPROVED** [review]

**Pull request merged**